src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass.cs

```diff
@@ -2768,6 +2768,52 @@ public override BoundNode VisitNonConstructorMethodBody(BoundNonConstructorMetho
             return null;
         }
 
+        public override BoundNode VisitNullCoalesingAssignmentOperator(BoundNullCoalesingAssignmentOperator node)
+        {
+            LocalState savedState;
+            if (RegularPropertyAccess(node.LeftOperand))
+            {
+                var left = (BoundPropertyAccess)node.LeftOperand;
+                var property = left.PropertySymbol;
+                if (property.RefKind == RefKind.None)
+                {
+                    var readMethod = property.GetOwnOrInheritedGetMethod() ?? property.SetMethod;
+                    var writeMethod = property.GetOwnOrInheritedSetMethod() ?? property.GetMethod;
+
+                    Debug.Assert(node.HasAnyErrors || (object)readMethod != (object)writeMethod);
+
+                    VisitReceiverBeforeCall(left.ReceiverOpt, readMethod);
+                    if (_trackExceptions) NotePossibleException(node);
+                    VisitReceiverAfterCall(left.ReceiverOpt, readMethod);
+
+                    savedState = this.State.Clone();
+
+                    VisitRvalue(node.RightOperand);
+                    PropertySetter(node, left.ReceiverOpt, writeMethod);
+
+                    ConditionallyAssignNullCoalescingOperator(node);
+
+                    IntersectWith(ref this.State, ref savedState);
+                    return null;
+                }
+            }
+
+            VisitRvalue(node.LeftOperand);
+            savedState = this.State.Clone();
+
+            VisitRvalue(node.RightOperand);
+            ConditionallyAssignNullCoalescingOperator(node);
+
+            IntersectWith(ref this.State, ref savedState);
```

Comment:
>IntersectWith [](start = 12, length = 13)

Are there tests for flow analysis? For instance:
```
object x;
object y;
G(x = 1).F ??= (y = 2);
x.ToString();
y.ToString(); // error
```

Reply:
Please include tests where using `IntersectWith` is observable, rather than just using `savedState`. That may require testing `DataFlowsOutWalker`.

---
In reply to: [199861695](https://github.com/dotnet/roslyn/pull/28253#discussion_r199861695) [](ancestors = 199861695)